### PR TITLE
Cleanup work in misc.cpp

### PR DESCRIPTION
**Pull request by mcostalba**
Also some code style tidy up of latest patches.

No functional change.

### CODE REVIEW

**Comment by glinscott** [comment]
Looks good!

File: src/search.cpp [hunk 1]
```diff
@@ -185,11 +185,11 @@ void Search::init() {
```
Comment (lucasart) [inline src/search.cpp]: While we're at it, can you change this comment

    /// Search::reset() clears all search memory, to restore a deterministic state

to:

    /// Search::reset() clears all search memory, to obtain reproducible search results

The word "deterministic" was a poor choice on my part. It means the relationship cause to consequence is a constant. Here it's different, we reset the cause to the same state, to guarantee the same consequence, so "reproducible" is a better word.

Comment (mcostalba) [inline src/search.cpp]: Done.

**Comment by zamar** [comment]
approved!

**Comment by mstembera** [comment]
I believe the change in tt.h is functional. The logic for saving eval16 is now different.  For example eval16 can now in some cases be written to even if ev == VALUE_NONE and the keys match.  Also eval16 will now not be written to in some cases even if the keys match and ev != VALUE_NONE.  These cases must be rare but if you run a deeper bench
stockfish.exe bench 16 1 22 default depth 1>nul
you will get Nodes searched:
before patch: 352189365
after patch: 355443730
I suspect the patch would likely pass a simplification test against the original but can't guarantee it.  It is a nice cleanup.

**Comment by mstembera** [comment]
I think logically eval16 should not depend on depth etc like value16 does.

**Comment by mstembera** [comment]
I instrumented the code and noticed that sometimes the new ev is slightly different from the already stored eval16 even though the keys match.  Implying that evaluate() sometimes produces different results for the same position?  I don't understand how/why this is possible.  Maybe we should discuss this separately from this patch though.

**Comment by glinscott** [comment]
@mstembera great catch!  there does seem to be a deeper issue here that we should investigate.

**Comment by zamar** [comment]
I think we should run [-4, 0] regression test at LTC for this patch. I
think it's unlikely that this patch could cause a major impact, but better
to be safe than sorry.

@mstembera: Feel free to schedule an extra test around this!

**Comment by mstembera** [comment]
eval16 should not depend on depth or bound.  If Marco or anyone has a reason why it should please let's hear it.  I have no problem w/ the rewrite itself so here is a non functional version of it.
```C++
  void save(Key k, Value v, Bound b, Depth d, Move m, Value ev, uint8_t g) {

    // Preserve any existing move for the same position
    if (m || (k >> 48) != key16)
        move16 = (uint16_t)m;
    
    if (ev != VALUE_NONE || (k >> 48) != key16)
        eval16 = (int16_t)ev;

    // Don't overwrite more valuable entries
    if (  (k >> 48) != key16
        || d > depth8 - 2
        || g != (genBound8 & 0xFC)
        || b == BOUND_EXACT)
    {
        key16     = (uint16_t)(k >> 48);
        value16   = (int16_t)v;
        genBound8 = (uint8_t)(g | b);
        depth8    = (int8_t)d;
    }
  }
```
No additional testing necessary since it is 100% functionally equivalent to the original.
I can open a separate pull request.

**Comment by mcostalba** [comment]
I am traveling with limited net access. I don't understand the reason to complicate current code.

**Comment by mstembera** [comment]
What do you not understand about the fact that eval16 does not depend on depth or bound but your change made it so anyway?
Enjoy your travel.  It can wait till you return.

**Comment by zamar** [comment]
I've scheduled a one-off regression test [-4, 0] at LTC with throughput of
500.

**Comment by mstembera** [comment]
I have no problem with the test itself. However, why should we insist on keeping code that was written as a mistake(the author didn't even know he was making a functional change)?  Not surprisingly that change doesn't make logical sense.  At least so far everyone has ignored my question why eval16 should depend on depth and bound?  All to save a single code line
"if (ev != VALUE_NONE || (k >> 48) != key16)".

**Comment by mstembera** [comment]
Also the test 
http://tests.stockfishchess.org/tests/view/555a546c0ebc5940ca5d6ba5
is scheduled w/ 128MB LTC which means zero effective hash pressure.  8MB would make sense to me.

**Comment by lucasart** [comment]
Indeed. And why LTC? STC with 4MB Hash would have been much more efficient and appropriate.

**Comment by mcostalba** [comment]
I understand the functional change, but the fact is that the code is simpler as is, so we could go the long route to add your patch then test for a simplification removing your patch with SPRT[-3, 1] and I am _very_ confident that it would pass. But in my opinon such approach would be a useless waste of resources becuase it is clear the cases where both patch differ are really rare.

**Comment by zamar** [comment]
The functional change in the patch is very minimalistic..

The purpose of my test is just to check that nothing is badly broken.

It's true that lower hash pressure would have make sense... Hmm... I'll
stop the current test and reschedule a new test...

**Pull request merged**